typewise_alert.py

```diff
@@ -1,46 +1,55 @@
-
-def infer_breach(value, lowerLimit, upperLimit):
-  if value < lowerLimit:
-    return 'TOO_LOW'
-  if value > upperLimit:
-    return 'TOO_HIGH'
-  return 'NORMAL'
-
+class cooling_type:
+    breach_type = 'Default'
+    def __init__(self,val,ul,ll):
+        self.infer_breach(val,ul,ll)
+    def infer_breach(self, value, upperLimit, lowerLimit):
+      if value < lowerLimit:
+        self.breach_type = 'TOO_LOW'
+      elif value > upperLimit:
+        self.breach_type = 'TOO_HIGH'
+      else:
+        self.breach_type = 'NORMAL'
 
 def classify_temperature_breach(coolingType, temperatureInC):
-  lowerLimit = 0
-  upperLimit = 0
-  if coolingType == 'PASSIVE_COOLING':
-    lowerLimit = 0
-    upperLimit = 35
-  elif coolingType == 'HI_ACTIVE_COOLING':
-    lowerLimit = 0
-    upperLimit = 45
-  elif coolingType == 'MED_ACTIVE_COOLING':
-    lowerLimit = 0
-    upperLimit = 40
-  return infer_breach(temperatureInC, lowerLimit, upperLimit)
-
-
-def check_and_alert(alertTarget, batteryChar, temperatureInC):
-  breachType =\
-    classify_temperature_breach(batteryChar['coolingType'], temperatureInC)
-  if alertTarget == 'TO_CONTROLLER':
-    send_to_controller(breachType)
-  elif alertTarget == 'TO_EMAIL':
-    send_to_email(breachType)
-
-
-def send_to_controller(breachType):
+    if coolingType == 'PASSIVE_COOLING':
+        return(cooling_type(temperatureInC,35,0).breach_type)
+    elif coolingType == 'HI_ACTIVE_COOLING':
+        return(cooling_type(temperatureInC,45,0).breach_type)
+    else:    
+       return(cooling_type(temperatureInC,40,0).breach_type)
+   
+def format_controller_message(header, breachType):
+    formatted_string = (f'{hex(header)}, {breachType}')
+    return formatted_string
+  
+def format_recepient(recepient):
+    formatted_string = (f'To: {recepient}')
+    return formatted_string
+    
+def print_message_on_console(message):
+    print(message)    
+    
+def send_to_controller(breachType,print_message_on_console):
   header = 0xfeed
-  print(f'{header}, {breachType}')
-
-
-def send_to_email(breachType):
+  controller_message = format_controller_message(header, breachType)
+  print_message_on_console(controller_message)
+  return controller_message
+    
+def send_to_email(breachType, print_message_on_console):
   recepient = "a.b@c.com"
```

Comment:
recipient is hardcoded. Atleast define a constant so that same can be used in test code too.